wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitExceptionMapper.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.git;
+
+import org.eclipse.che.api.git.exceptions.GitConflictException;
+import org.eclipse.che.api.git.exceptions.GitInvalidRefNameException;
+import org.eclipse.che.api.git.exceptions.GitRefAlreadyExistsException;
+import org.eclipse.che.api.git.exceptions.GitRefNotFoundException;
+import org.eclipse.che.dto.server.DtoFactory;
+import javax.inject.Singleton;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * Exception mapper for all GIT exceptions
+ *
+ * @author Yossi Balan ( yossi.balan@sap.com)
+ */
+@Provider
+@Singleton
+public class GitExceptionMapper implements ExceptionMapper<GitException> {
+
+    /**
+     * check the exception type and build Response with the status body and the type of the error
+     */
+    @Override
+    public Response toResponse(GitException exception) {
+        if (exception instanceof GitRefNotFoundException)
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(DtoFactory.getInstance().toJson(exception.getServiceError()))
+                    .type(MediaType.APPLICATION_JSON).build();
+        else if (exception instanceof GitInvalidRefNameException)
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(DtoFactory.getInstance().toJson(exception.getServiceError()))
+                    .type(MediaType.APPLICATION_JSON).build();
+        else if (exception instanceof GitRefAlreadyExistsException)
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(DtoFactory.getInstance().toJson(exception.getServiceError()))
+                    .type(MediaType.APPLICATION_JSON).build();
+        else if (exception instanceof GitConflictException) {
+            ConflictExceptionError conflictExceptionError = DtoFactory.getInstance()
+                    .createDto(ConflictExceptionError.class)
```

Comment:
![screenshot from 2016-07-21 13-41-57](https://cloud.githubusercontent.com/assets/7668752/17020267/4c532894-4f49-11e6-9513-b892970e75b2.jpg)

That is because DTO generator in che-core-api-git configured to take DTO from org.eclipse.che.api.git.shared. You should move `ConflictExceptionError` there and this should work